### PR TITLE
Modernize the model tests

### PR DESCRIPTION
**Pull request by sambuddhabasu**

### CODE REVIEW

**Comment by preddy5** [comment]
:+1:

File: bookie/tests/test_models/test_bmark.py [hunk 1]
```diff
@@ -8,28 +7,13 @@
...
 from bookie.models.auth import User
 
-from bookie.tests import empty_db
 from bookie.tests import gen_random_word
-from bookie.tests import TestDBBase
+from bookie.tests import TestViewBase
 
 
-class TestBmark(TestDBBase):
+class TestBmark(TestViewBase):
```
Comment (mitechie) [inline bookie/tests/test_models/test_bmark.py]: why is this TestViewBase? There's no api or web request test using webtest in here.

File: bookie/tests/test_models/test_bmarkmgr.py [hunk 2]
```diff
@@ -11,28 +10,13 @@
 )
 from bookie.models.auth import User
 
-from bookie.tests import empty_db
 from bookie.tests import gen_random_word
-from bookie.tests import TestDBBase
+from bookie.tests import TestViewBase
```
Comment (mitechie) [inline bookie/tests/test_models/test_bmarkmgr.py]: Again, the model tests should work with just TestDBBase vs ViewBase. That's specifically for setting up the app for api/html requests.

File: bookie/tests/test_models/test_privatebmark.py [hunk 3]
```diff
@@ -8,28 +7,13 @@
 )
 from bookie.models.auth import User
 
-from bookie.tests import empty_db
 from bookie.tests import gen_random_word
-from bookie.tests import TestDBBase
+from bookie.tests import TestViewBase
```
Comment (mitechie) [inline bookie/tests/test_models/test_privatebmark.py]: same question

File: bookie/tests/test_models/test_tagmgr.py [hunk 4]
```diff
@@ -1,43 +1,19 @@
...
-    make_tag,
-)
-
-import os
+from bookie.tests import TestViewBase
+from bookie.tests.factory import make_tag
 
 
-class TestTagMgrStats(TestDBBase):
+class TestTagMgrStats(TestViewBase):
```
Comment (mitechie) [inline bookie/tests/test_models/test_tagmgr.py]: same thing, you've removed the View based tests from here.

File: bookie/tests/__init__.py [hunk 5]
```diff
@@ -65,6 +65,8 @@ def gen_random_word(wordLen):
 class TestDBBase(unittest.TestCase):
     def setUp(self):
         """Setup Tests"""
+        from bookie.models import initialize_sql
```
Comment (mitechie) [inline bookie/tests/__init__.py]: I think this can actually be done before here after line 37

File: bookie/tests/test_api/test_base_api.py [hunk 6]
```diff
@@ -8,15 +8,22 @@
...
 from bookie.models.auth import Activation
 from bookie.tests import BOOKIE_TEST_INI
 from bookie.tests import empty_db
 from bookie.tests import factory
 from bookie.tests import gen_random_word
+from bookie.tests.factory import make_bookmark
 
 from datetime import datetime
 
+import os
```
Comment (mitechie) [inline bookie/tests/test_api/test_base_api.py]: importing OS should be up at the top of the import list as it's a built in. It should come before all booke.\* imports.

**Comment by mitechie** [comment]
Looks good with the 2 small changes noted. If tests pass with those changes :+1: to land.

**Comment by sambuddhabasu** [comment]
:shipit:

**Pull request merged**